okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/CurrentUrlInterceptor.java

```diff
@@ -44,7 +44,7 @@ public Response intercept(Chain chain) throws IOException {
                 .url(rebasedUrl)
                 // Request.this.tag field by default points to request itself if it was not set in RequestBuilder.
                 // We don't want to reference old request in new one - that is why we need to reset tag to null.
-                .tag(originalRequest.tag().equals(originalRequest) ? null : originalRequest.tag())
+                .tag(originalRequest.equals(originalRequest.tag()) ? null : originalRequest.tag())
```

Comment:
was this a bug? if so, can you make it a separate PR, please?

Reply:
no, they broke

Reply:
tags are now a map keyed by Class, and the old one is the Object case

Reply:
https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/Request.java#L137

Reply:
vs https://github.com/square/okhttp/blob/parent-3.9.0/okhttp/src/main/java/okhttp3/Request.java#L42

Reply:
and it returns null

Reply:
@jkozlowski this whole block should no longer be necessary, looking at the new code - don't need to unset the tag

Reply:
but maybe we want it to continue working the way it does for people who force their okhttp version, and so this'd be fine

Reply:
cool ignore me :)

Reply:
Soo, here's my understanding: in 3.9.0 the Request.Builder#tag method had the following javadoc:

```
    /**
     * Attaches {@code tag} to the request. It can be used later to cancel the request. If the tag
     * is unspecified or null, the request is canceled by using the request itself as the tag.
     */
```

The value, as @j-baker pointed out, is automatically set in Request(Builder) constructor to Request.this, and then propagated across to the next builder when the request is cloned (e.g. for retries). This means it builds a linked list of requests which means they all linger on in memory.

It would seem to me this was an omission and not intentional behavior: this is now gone in OkHttp 3.11 so I think we are good to remove this.

Reply:
Going to resolve this point, unless people object.